### ISSUE TITLE
open-last-project disable "new file" feature

### ISSUE THREAD

**Issue by TouPye** [danielmahon/atom-open-last-project#28]
According to [this issue](https://github.com/atom/atom/issues/5648) I cretaed on the main Atom project, it seems like `open-last-project` disable the ability to create a new file.

When I disabled this package, it fixed my issue.

**Comment by TouPye** [comment]
I rollbacked my package, replacing main.coffee and open-last-project.coffee files by their previous version and it now works fine.

**Comment by steelbrain** [comment]
Can you confirm this issue after opening
the dev tools (CTRL+SHIFT+I or Top Toolbar View->Developer->Toggle Developer Mode) and then running `localStorage.removeItem('open-last-project')` and restarting fixes your problem?

**Comment by jeffmcneill** [comment]
I've got the same problem, namely cannot create a new file with Atom 182 + 0.3.1 open-last-project. Just got an error when trying to create a new file. Disable open-last-project and can create files again (but not remember the project tabs).

**Comment by steelbrain** [comment]
@jeffmcneill That's interesting, I've been using Open-Last-Project for a lot of time without a problem, Can you confirm that you have the same problem with Atom v0.179?
Atom itself has been having some problems since v0.180

**Comment by jeffmcneill** [comment]
I've tried with Atom v0.179 and the same problem with open-last-project 0.3.1, I cannot create a new file. I moved back to 0.2.2 and it doesn't work (but I can create new files). Tried launching from browser and command line. 

Attaching screenshots of 0.3.1 vs. 0.2.2 file differences:
![screen shot 2015-02-22 at 10 28 54 pm](https://cloud.githubusercontent.com/assets/3119827/6319094/027c07fa-bae3-11e4-875d-f3832c71b462.png)

**Comment by steelbrain** [comment]
@jeffmcneill Do you see any sort of error or something in Atom or the developer console?
Normally if an event handler throws error, it aborts further execution of the event.

**Comment by steelbrain** [comment]
@jeffmcneill @TouPye I've pushed a possible fix in my repo, They are in the Pull Request #27
Because I wasn't able to reproduce the issue myself, It would be great if you would try it for me and tell me how it goes
```sh
cd .atom/packages
rm -rf open-last-project
git clone https://github.com/steelbrain/atom-open-last-project open-last-project
```
Then Open Atom and try to create a new file again.

**Comment by jeffmcneill** [comment]
@steelbrain @TouPye I'm going to follow the directions from benogle where he discusses atom already remembering tabs when launching from directories with command atom .

see: https://github.com/atom/atom/issues/1603

This should be in core at some point.

**Comment by steelbrain** [comment]
@jeffmcneill He doesn't really discuss about atom remembering tabs, He discusses about Atom's [De]Serialization manager, Where you save your serialization state on exit and when you open atom within a directory, that directory's serialization state is then given back to the packages allowing them to re-open the same files.
I gave it a try once, but couldn't use it because it only works when you launch atom from that directory, and when you set the project path programatically (that's what open-last-project) does, You don't get that info, making it unusable.

**Comment by TouPye** [comment]
@steelbrain I tried what you ask and here are the resultst :

In the dev tool, when I input `localStorage.removeItem('open-last-project')` console returns me `undefined` then when I restart Atom, **nothing have changed**.

Then, I cloned your repo as you said. **I can create a new file, but it doesn't save the last project, and I can't open a new folder** (treeview is locked on `C:\Users\Pierre\AppData\Local\atom`)

Once again, I **rollbacked** `main.coffee` and `open-last-project.coffee` to the (currently) 4days ago commit (called "Revert use of Atom Serialization") and **it works perfectly**.

**Comment by steelbrain** [comment]
@TouPye What If I told you that it's the latest commit in this repo, You only rolled back the package.json's version and no functionality.
This is super weird.

**Comment by TouPye** [comment]
@steelbrain I'm talking about [this one](https://github.com/danielmahon/atom-open-last-project/commit/9c488a204ab8610963c526366e3c0f9494879626)

**Comment by steelbrain** [comment]
@TouPye The only API related change after that is `atom.project.setPath` instead of `atom.project.setPaths` https://github.com/danielmahon/atom-open-last-project/pull/27/files#diff-6a556ba425d22e307675101c5806db06R19

**Comment by steelbrain** [comment]
Guys I think I just fixed it. Can someone confirm by checking out HEAD of my repo?

**Comment by danielmahon** [comment]
Im checking it out right now

**Comment by danielmahon** [comment]
this is still a problem, new files seems to be working but new file in new window activates open-last-project which it shouldnt, we need to check for something other than just `atom.project.getPath`

**Comment by steelbrain** [comment]
@danielmahon It's actually an Atom thing, They're working on adding support for multiple folders to tree view.
and Because it's a bit unstable at the moment, I am not sure what the problem. I added some verbosity and I found out that `loadProject` is not really executed when a new window is opened.
and about that `atom.project.getPath` I have added a timeout to it now so it'll wait for atom to initialize properly.
